subprojects/plugins/src/test/groovy/org/gradle/api/plugins/JavaPluginTest.groovy

```diff
@@ -199,6 +201,8 @@ class JavaPluginTest extends AbstractProjectBuilderSpec {
         def javaLibrary = project.components.getByName("java")
 
         then:
+        javaLibrary instanceof JvmSoftwareComponent
+        javaLibrary instanceof AdhocComponentWithVariants
```

Comment:
💭 Seeing the two tests like this really makes think the class hierarchy needs to be updated.